rfcs/20190802-model-garden-redesign.md

```diff
@@ -0,0 +1,207 @@
+|               | utils                |   |                                 |
+|               | examples             |   | reference models for testing/validating end-to-end tensforflow |
+|               |                      | Resnet |                            |
+|               |                      | BERT   |                            |
+|               |                      | Transformer |                       |
+|               |                      | NCF   |                             |
+| nlp           |                      |   | models/tasks for Natural Language Processing |
+|               | utils                |   | NLP specific utils, e.g. input dataset |
+|               | BERT                 |   | NLP specific utils, e.g. input dataset |
+|               |                      | BERT core modeling |  |
+|               |                      | tasks | specific tasks on open public datasets, e.g Squad, MNLI |
+|               | XLNET                |    |   |
+|               | GPT                  |    |   |
+|               | Transformer          |    |   |
+|               | GNMT                 |    |   |
+|               | ...                  |    |   |
+| vision        |                      |    | models/tasks for Computer Vision |
+|               | image_classification |    |   |
+|               |                      | resnet |   |
+|               |                      | EfficientNet   |   |
+|               |                      | MnasNet   |   |
+|               |                      | ...   |   |
+|               | object_detection     |    |   |
+|               |                      | RetinaNet    |   |
+|               |                      | Mask-RCNN    |   |
+|               |                      | UNet    |   |
+|               |                      | ShapeMask    |   |
+|               |                      | ...    |   |
+| recommendation|                      |     |   |
+|               | NCF                  |     |   |
+| staging       |                      |     | accepting community contributions  |
+| archive       |                      |     | deprecated models, not officially supported  |
+| r1            |                      |     | tf1.x models and utils  |
+|               | utils                |     |   |
+|               | resnet50             |     |   |
+|               | transformer          |     |   |
+|               | ncf                  |     |   |
+|               | wide_deep            |     |   |
+|               | boosted_trees        |     |   |
+
+### Pretrained model repository
+
+We are going to provide the pretrained models for research exploration and
+real-world application development. The plan is to integrate with [TensorFlow Hub](https://www.tensorflow.org/hub),
+where users can access the Hub modules and SavedModel for pretrained checkpoints and links to the code in the model
+garden.
```

Comment:
Please avoid use of `configs` as much as possible. The code should be flexible enough to change the parameters in the code itself. Changing things though a `config` file makes sense for some people but not for all. It is easy to maintain the changes in the code but it is hard to track the changes in the config file and at the same time, it is hard to validate the changes as well

Reply:
Thanks for your suggestion! We plan to follow the practice in cloud tpu model repo, for example, https://github.com/tensorflow/tpu/blob/master/models/official/mnasnet/mnasnet_main.py#L674. A default parameter dictionary will be provided, but users can overwrite the default value or extend to new key/value with string, dict or yaml.